File: programming/problem_6/problem_6.py
from typing import List


def average_excluding_min_max(input_numbers: List[int]) -> float:
    min_val = min(input_numbers)
    max_val = max(input_numbers)
    filtered_numbers = [e for e in input_numbers if e != min_val and e != max_val]
    sum_val = sum(filtered_numbers)
    return sum_val / len(filtered_numbers)


print("average_excluding_min_max:", average_excluding_min_max([1, 2, 3, 4, 5]))

File: programming/problem_6/problem_6.kt
fun averageExcludingMinMax(inputNumbers: List<Int>): Double {
    val min = inputNumbers.minOrNull() ?: 0
    val max = inputNumbers.maxOrNull() ?: 0
    val filteredNumbers = inputNumbers.filter { it != min && it != max }
    val sum = filteredNumbers.sum()
    return sum.toDouble() / filteredNumbers.size
}

fun main() {
    println("averageExcludingMinMax: " + averageExcludingMinMax(listOf(1, 2, 3, 4, 5)))
}
